src/main/java/randoop/sequence/VariableRenamer.java

```diff
@@ -61,52 +73,113 @@ static String getVariableName(Type type) {
+      if (type.equals(JavaTypes.FLOAT_TYPE)) {
+        return "float";
+      }
+      if (type.equals(JavaTypes.DOUBLE_TYPE)) {
+        return "double";
+      }
+      if (type.equals(JavaTypes.SHORT_TYPE)) {
+        return "short";
+      }
+
+      // Otherwise, use the first character of the type name.
       return type.getName().substring(0, 1);
     }
 
-    // special cases:  Object, String, Class
-    if (type.isObject()) {
-      return "obj";
-    } else if (type.isString()) {
-      return "str";
-    } else if (type.equals(JavaTypes.CLASS_TYPE)) {
-      return "cls";
-    }
-    // TODO: add more special cases:
-    //  * if it implements List, use "list"
-    //  * if it implements Iterator, use "itor"
-    // ... more?
+    // Get the simple name of the type.
+    String varName = type.getSimpleName();
 
     if (type.isParameterized()) {
-      ClassOrInterfaceType classType = (ClassOrInterfaceType) type;
-      String varName = classType.getSimpleName().toLowerCase();
-      if (varName.equals("class")) {
+      if (varName.toLowerCase().equals("class")) {
         return "cls";
       }
-      for (TypeArgument argument : classType.getTypeArguments()) {
-        if (argument.isWildcard()) {
-          varName += "_" + "wildcard";
-        } else {
-          String argumentName = getVariableName(((ReferenceArgument) argument).getReferenceType());
-          varName += "_" + argumentName;
+
+      // Special cases for parameterized types.
+      if (varName.contains("Iterator")) {
+        // Iterator takes precedence, in cases like ListIterator.
+        varName = "itor";
+      } else if (varName.contains("List")) {
+        // List comes before array, in cases like ArrayList.
+        varName = "list";
+      } else if (varName.contains("Set")) {
+        varName = "set";
+      } else if (varName.contains("Map")) {
+        varName = "map";
+      } else if (varName.contains("Queue")) {
+        varName = "queue";
+      } else if (varName.contains("Collection")) {
+        varName = "collection";
+      } else if (varName.contains("Array")) {
+        varName = "array";
+      }
+
+      // Only use the first type argument to construct the name to simplify things.
+      ClassOrInterfaceType classType = (ClassOrInterfaceType) type;
+      TypeArgument argument = classType.getTypeArguments().get(0);
+      if (argument.isWildcard()) {
+        varName = "wildcard" + capitalizeString(varName);
+      } else {
+        if (depth < VAR_NAME_MAX_DEPTH) {
+          String argumentName =
+              getVariableName(((ReferenceArgument) argument).getReferenceType(), depth + 1);
+
+          varName = argumentName + capitalizeString(varName);
+        }
+      }
+    } else {
+      // Special cases: Object, String, Class.
+      if (type.isObject()) {
+        varName = "obj";
+      } else if (type.isString()) {
+        varName = "str";
+      } else if (type.equals(JavaTypes.CLASS_TYPE)) {
+        varName = "cls";
```

Comment:
This is the second occurrence of a special case for `Class` and `cls`.  Remove one of them.